include/z64save.h

```diff
@@ -273,6 +273,13 @@ typedef enum SunsSongState {
     /* 3 */ SUNSSONG_SPECIAL // time does not advance, but signals the song was played. used for freezing redeads
 } SunsSongState;
 
+typedef enum DungeonIndex {
+    /* 0 */ DUNGEON_INDEX_WOODFALL_TEMPLE,
+    /* 1 */ DUNGEON_INDEX_SNOWHEAD_TEMPLE,
+    /* 2 */ DUNGEON_INDEX_GREAT_BAY_TEMPLE,
+    /* 3 */ DUNGEON_INDEX_STONE_TOWER_TEMPLE, // Also applies to Inverted Stone Tower Temple
+} DungeonIndex;
```

Comment:
```suggestion
typedef enum {
    /* 0 */ DUNGEON_INDEX_WOODFALL_TEMPLE,
    /* 1 */ DUNGEON_INDEX_SNOWHEAD_TEMPLE,
    /* 2 */ DUNGEON_INDEX_GREAT_BAY_TEMPLE,
    /* 3 */ DUNGEON_INDEX_STONE_TOWER_TEMPLE, // Also applies to Inverted Stone Tower Temple
} DungeonIndex;
```
Don't need this I believe. Also, this probably makes more sense in `z64save.h` since it is an enum for `gSaveContext.dungeonIndex`. But I should at least ask if it's worth moving to `z64map.h`. Don't think so though

Reply:
I think it belongs in `z64save.h,` yeah. In a future where we don't just combine everything into one big header, someone who only interacts with `gSaveContext` probably expects to get this enum when they include `z64save.h`.